File: ComputerPartsEcommerce-ejb/src/java/entity/ComputerPart.java
package entity;

import java.io.Serializable;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;

@Entity
@Inheritance(strategy=InheritanceType.JOINED)
public class ComputerPart extends Product implements Serializable {

    private static final long serialVersionUID = 1L;

    public ComputerPart() {
    }

    public ComputerPart(String name, Double price, Integer inventoryQuantity, String image) {
        super(name, price, inventoryQuantity, image);
    }

}

File: ComsmanCustomerClientAngular/src/app/computer-part.ts
import { Product } from './product';

export class ComputerPart extends Product {

    constructor(productId?: number, name?: string, price?: number, inventoryQuantity?: number, image?: string, isDisabled?: boolean) {
        super(productId, name, price, inventoryQuantity, image, isDisabled);
    }
}
